frontend/packages/topology/src/filters/FilterProvider.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import { useUserSettings } from '@console/shared';
+import { getDefaultAppliedFilters } from '../redux/reducer';
+import { DEFAULT_TOPOLOGY_FILTERS } from './const';
+import { getAppliedFilters } from '../redux/action';
+import { DisplayFilters } from '../topology-types';
+
+const TOPOLOGY_DISPLAY_FILTERS_CONFIG_MAP_KEY = `devcosole.topology.filters`;
+
+const getTopologyFilters = (appliedFilters: AppliedFilters) => {
+  const filters = [...DEFAULT_TOPOLOGY_FILTERS];
+  filters.forEach((filter) => {
+    if (appliedFilters[filter.id] !== undefined) {
+      filter.value = appliedFilters[filter.id];
+    }
+  });
+
+  return filters;
+};
+
+type AppliedFilters = { [filterKey: string]: boolean };
+type SetTopologyFilters = (filters: DisplayFilters) => void;
+
+const useFilterContextValues = (): [
+  DisplayFilters,
+  AppliedFilters,
+  boolean,
+  SetTopologyFilters,
+] => {
+  const [appliedFilters, setAppliedFilters, appliedFiltersLoaded] = useUserSettings(
+    TOPOLOGY_DISPLAY_FILTERS_CONFIG_MAP_KEY,
+    getDefaultAppliedFilters(),
+  );
+  const [filtersLoaded, setFiltersLoaded] = React.useState<boolean>(false);
+  const [filters, setFilters] = React.useState<DisplayFilters>([]);
+
+  React.useEffect(() => {
+    if (appliedFiltersLoaded && !filtersLoaded) {
+      setFilters(getTopologyFilters(appliedFilters));
+      setFiltersLoaded(true);
+    }
+  }, [appliedFilters, appliedFiltersLoaded, filtersLoaded]);
+
+  const setTopologyFilters = React.useCallback(
+    (displayFilters: DisplayFilters) => {
+      setFilters(displayFilters);
+      setAppliedFilters(getAppliedFilters(displayFilters));
+    },
+    [setAppliedFilters],
+  );
+
+  return [filters, appliedFilters, appliedFilters && filtersLoaded, setTopologyFilters];
```

Comment:
If `appliedFilters` are what we have stored in user settings... and `filters` are just `appliedFilters` with added defaults... why are we providing both back? Wouldn't we want just `filters`?

Reply:
Because `appliedFilters` is also being used in component.

Reply:
Hmmm... we probably should look to refactor that -- let's go with this for now since it's the status quo with the implementation. But it seems like an odd situation.